GPU/GPUTracking/Merger/GPUTPCGMPhysicalTrackModel.cxx

```diff
@@ -153,6 +153,7 @@ GPUd() int GPUTPCGMPhysicalTrackModel::PropagateToXBxByBz(float x, float Bx, flo
   if (t.PropagateToXBzLightNoUpdate(xe, bb, dLp) != 0) {
     return -1;
   }
+  // TODO Should one not call t.UpdateValues() here?
```

Comment:
@sgorbuno : can you comment if UpdateValues() is needed here.

Reply:
Sorry, this check I should have done before committing. UpdateValues does not affect the physical parameters of the trajectory (x,y,z,px,py,pz). The function does not need to be called between two steps of PropagateToXBzLightNoUpdate, because in there only the physical parameters are touched. So I will remove this line.

Reply:
Hi Ole, David, 

UpdateValues() is a confusing name..  It should not be called.

It updates some complementary values, which are calculated out of the track parameters. 
For example, Pt is calculated out of (Px,Py,Pz), and saved as a class member.
These complimentary values are not used in extrapolation & rotation, therefore one can update them only at the very end of the extrapolation, thus saving the CPU time.

Reply:
Thanks for the confirmation and explanation! I have dropped the comment in the code.